### PR TITLE
BUG: Fix all-masked imshow

### PR DESCRIPTION
**Pull request by larsoner**
## PR Summary

Deal with `imshow`ing an array that is completely masked without emitting a warning. The added test/code emits 3 warnings on `master`, none of which are particularly informative. The test on `master` fails with:

<details>

```
lib/matplotlib/image.py:639: in draw
    im, l, b, trans = self.make_image(
lib/matplotlib/image.py:924: in make_image
    return self._make_image(self._A, bbox, transformed_bbox, clip,
lib/matplotlib/image.py:476: in _make_image
    vrange = np.array([vmin, vmax], dtype=scaled_dtype)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = masked

    def __float__(self):
        """
        Convert to float.
    
        """
        if self.size > 1:
            raise TypeError("Only length-1 arrays can be converted "
                            "to Python scalars")
        elif self._mask:
>           warnings.warn("Warning: converting a masked element to nan.", stacklevel=2)
E           UserWarning: Warning: converting a masked element to nan.
```

</details>

And on latest `NumPy` it also emits a warning later having to do with `np.clip` not liking `np.nan` values.

Alternatively, `imshow` with all data masked could emit a warning, but it seems like the existing code had a chance to do that but didn't (there is an existing `if a_min is np.ma.masked` on line 409) so I didn't implement it here.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [x] New features are documented, with examples if plot related.
- [x] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and `pydocstyle<4` and run `flake8 --docstring-convention=all`).
- [x] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [x] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

**Review by tacaswell — APPROVED** [review]

**Review by QuLogic — APPROVED** [review]

**Comment by larsoner** [comment]
Thanks for the quick reviews!

File: lib/matplotlib/tests/test_image.py [hunk 1]
```diff
@@ -853,6 +853,14 @@ def test_mask_image():
     ax2.imshow(A, interpolation='nearest')
 
 
+def test_mask_image_all():
+    # Test behavior with an image that is entirely masked does not warn
+    data = np.full((2, 2), np.nan)
+    fig, ax = plt.subplots()
+    ax.imshow(data)
+    fig.canvas.draw_idle()  # would emit a warning
```
Comment (dopplershift) [inline lib/matplotlib/tests/test_image.py:861]: Unless we run with warnings as error (which I'm pretty sure we don't) this test doesn't actually check that we don't issue warnings...

Comment (larsoner) [reply lib/matplotlib/tests/test_image.py:861]: I think it failed on master before the other changes in this PR, and this suggested to me that you do:

https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/testing/conftest.py#L20

But I can wrap this in a `with pytest.warns(None) as w: ... assert len(w) == 0` if it'll help

Comment (larsoner) [reply lib/matplotlib/tests/test_image.py:861]: (see the traceback in the description, that was from the original test on `master` without the fixing-changes)

Comment (QuLogic) [reply lib/matplotlib/tests/test_image.py:861]: Pytest does treat warnings as errors.

Comment (dopplershift) [reply lib/matplotlib/tests/test_image.py:861]: Ah, I wasn't aware we had that turned on in `conftest.py`. I'm used to testing on MetPy where so many upstream dependencies have warnings at times that running with warnings is impractical. Apologies for the noise.

**Pull request merged**